testsuite/MDAnalysisTests/lib/test_augment.py

```diff
@@ -0,0 +1,100 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding:utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4 fileencoding=utf-8
+#
+# MDAnalysis --- https://www.mdanalysis.org
+# Copyright (c) 2006-2017 The MDAnalysis Development Team and contributors
+# (see the file AUTHORS for the full list of names)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# R. J. Gowers, M. Linke, J. Barnoud, T. J. E. Reddy, M. N. Melo, S. L. Seyler,
+# D. L. Dotson, J. Domanski, S. Buchoux, I. M. Kenney, and O. Beckstein.
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+
+from __future__ import absolute_import
+from six.moves import zip
+import pytest
+import numpy as np
+from numpy.testing import assert_almost_equal, assert_equal
+from itertools import product
+
+from MDAnalysis.lib._augment import augment_coordinates, undo_augment
+from MDAnalysis.lib.distances import apply_PBC, transform_StoR
+
+boxes = ([10, 10, 10, 90, 90, 90],  # ortho
+         [10, 10, 10, 45, 60, 90])  # tri_box
+
+# Find images for several query points,
+# here in fractional coordinates using augment
+queries = ([0.1, 0.5, 0.5],  # box face
+           [0.5, 0.5, 0.5],  # box center
+           [0.5, -0.1, 0.5],  # box face
+           [0.1, 0.1, 0.5],  # box edge
+           [0.5, -0.1, 1.1],  # box edge
+           [0.1, 0.1, 0.1],  # box vertex
+           [0.1, -0.1, 1.1],  # box vertex
+           [2.1, -3.1, 0.1],  # box vertex
+           [[0.1, 0.5, 0.5],
+            [0.5, -0.1, 0.5]]  # Multiple vectors
+           )
+
+
+# Images for the previous query vectors, here in fractional coordinates.
+images = (([1.1, 0.5, 0.5],),
+          (),
+          ([0.5, -0.1, 0.5],),
+          ([1.1, 0.1, 0.5], [0.1, 1.1, 0.5], [1.1, 1.1, 0.5]),
+          ([0.5, -0.1, 0.1], [0.5, 0.9, 1.1], [0.5, -0.1, 1.1]),
+          ([1.1, 0.1, 0.1], [0.1, 1.1, 0.1], [0.1, 0.1, 1.1],
+           [0.1, 1.1, 1.1], [1.1, 1.1, 0.1], [1.1, 0.1, 1.1],
+           [1.1, 1.1, 1.1]),
+          ([1.1, 0.9, 0.1], [0.1, -0.1, 0.1], [0.1, 0.9, 1.1],
+           [0.1, -0.1, 1.1], [1.1, -0.1, 0.1], [1.1, 0.9, 1.1],
+           [1.1, -0.1, 1.1]),
+          ([1.1, 0.9, 0.1], [0.1, -0.1, 0.1], [0.1, 0.9, 1.1],
+           [0.1, -0.1, 1.1], [1.1, -0.1, 0.1], [1.1, 0.9, 1.1],
+           [1.1, -0.1, 1.1]),
+          ([1.1, 0.5, 0.5], [0.5, -0.1, 0.5]))
+
+radius = 1.5
+
+
+@pytest.mark.parametrize('b, qres', product(boxes, zip(queries, images)))

```

Comment:
don't use `product`, just have two `mark.parametrize`, one for `b` one for `qres`